### ISSUE TITLE
GET requests on forms do not overwrite URL parameters

### ISSUE THREAD

**Issue by faissalMT** [teamcapybara/capybara#2311]
**Note:** This is for issues with Capybara.  If you have a howto type question, please ask on the mailing list as requested in the README: http://groups.google.com/group/ruby-capybara

## Meta
Capybara Version:
3.29.0
Driver Information (and browser if relevant):
`rack_test`

## Expected Behavior
The query params in the form target URL should be replaced. This is what both Chrome and Firefox do.  

## Actual Behavior
The query params in the form target URL are appended to. The only browser I can find that behaves this way is Links2.

## Steps to reproduce
In a view:
```
<%= form_with(url: "https://www.google.com/search?source=test", method: "get") do %>
  <%= f.text_field(:q, id: "query") %>
<% end %>
```
In a test for said page:
```
page.find("#query").fill_in("my search query")
```

**Comment by twalpole** [comment]
I believe the current behavior was correct for HTML4 - but isn't for HTML5

**Comment by twalpole** [comment]
Fixed via PR #2312
